hls-graph/src/Development/IDE/Graph/Internal/Database.hs

```diff
@@ -133,26 +133,40 @@ builder db@Database{..} stack keys = withRunInIO $ \(RunInIO run) -> do
                 waitAll
                 pure results
 
+isDirty :: Foldable t => Result -> t (a, Result) -> Bool
+isDirty me = any (\(_,dep) -> resultBuilt me < resultChanged dep)
+
+-- | Refresh dependencies for a key and compute the key:
+-- The deps refresh is kicking up linearly. If any of the deps are dirty in the process,
```

Comment:
Can you please explain what does kicking mean in this context?
Also it's not clear to me what you mean by linearly. You're referring to linear order of something, but what's not clear to me is: linear order "of what"?

Reply:
we used to try to try to kick all the deps concurrently and collect all the results. The linearly means one following another. Following the deps order in the rule.

Reply:
Kicking means try to compute the result, if we have clean cache, it might be just a look up.